pkg/nutanix/v3/v3_structs.go

```diff
@@ -160,261 +168,292 @@ type GuestCustomizationSysprep struct {
 
-	PowerStateMechanism *VMPowerStateMechanism `json:"power_state_mechanism,omitempty"`
+	PowerStateMechanism *VMPowerStateMechanism `json:"power_state_mechanism,omitempty" mapstructure:"power_state_mechanism,omitempty"`
 
 	// Indicates whether VGA console should be enabled or not.
-	VgaConsoleEnabled *bool `json:"vga_console_enabled,omitempty"`
+	VgaConsoleEnabled *bool `json:"vga_console_enabled,omitempty" mapstructure:"vga_console_enabled,omitempty"`
+
+	// Indicates whether to passthrough the host’s CPU features to the guest. Enabling this will disable live migration of the VM.
+	EnableCPUPassthrough *bool `json:"enable_cpu_passthrough,omitempty" mapstructure:"enable_cpu_passthrough,omitempty"`
+
+	// Indicates whether a VMs cores are getting pinned on either CPU1, CPU2, CPU3 or CPU4. By default, the Linux Scheduler in AHV will pin all cores wherever they are best available.
+	EnableCPUPinning *bool `json:"is_vcpu_hard_pinned,omitempty" mapstructure:"is_vcpu_hard_pinned,omitempty"`
 
 	// Information regarding vNUMA configuration.
-	VMVnumaConfig *VMVnumaConfig `json:"vnuma_config,omitempty"`
+	VMVnumaConfig *VMVnumaConfig `json:"vnuma_config,omitempty" mapstructure:"vnuma_config,omitempty"`
+
+	SerialPortList []*VMSerialPort `json:"serial_port_list,omitempty" mapstructure:"serial_port_list,omitempty"`
 
-	SerialPortList []*VMSerialPort `json:"serial_port_list,omitempty"`
+	MachineType *string `json:"machine_type,omitempty" mapstructure:"machine_type,omitempty"`
 }
 
 // VM An intentful representation of a vm spec
 type VM struct {
-	AvailabilityZoneReference *Reference `json:"availability_zone_reference,omitempty"`
+	AvailabilityZoneReference *Reference `json:"availability_zone_reference,omitempty" mapstructure:"availability_zone_reference,omitempty"`
 
-	ClusterReference *Reference `json:"cluster_reference,omitempty"`
+	ClusterReference *Reference `json:"cluster_reference,omitempty" mapstructure:"cluster_reference,omitempty"`
 
 	// A description for vm.
-	Description *string `json:"description,omitempty"`
+	Description *string `json:"description,omitempty" mapstructure:"description,omitempty"`
 
 	// vm Name.
-	Name *string `json:"name"`
+	Name *string `json:"name" mapstructure:"name"`
 
-	Resources *VMResources `json:"resources,omitempty"`
+	Resources *VMResources `json:"resources,omitempty" mapstructure:"resources,omitempty"`
 }
 
 // VMIntentInput ...
 type VMIntentInput struct {
-	APIVersion *string `json:"api_version,omitempty"`
+	APIVersion *string `json:"api_version,omitempty" mapstructure:"api_version,omitempty"`
 
-	Metadata *Metadata `json:"metadata"`
+	Metadata *Metadata `json:"metadata" mapstructure:"metadata"`
 
-	Spec *VM `json:"spec"`
+	Spec *VM `json:"spec" mapstructure:"spec"`
 }
 
 // MessageResource ...
 type MessageResource struct {
 
 	// Custom key-value details relevant to the status.
-	Details interface{} `json:"details,omitempty"`
+	Details interface{} `json:"details,omitempty" mapstructure:"details,omitempty"`
```

Comment:
This type was changed from `map[string]string` to `interface{}` in https://github.com/nutanix-cloud-native/prism-go-client/commit/cc121d3d3c270c9d90419b8f9d0e19ef200f5702